lib/synapse/service_watcher/base.rb

```diff
@@ -55,13 +49,28 @@ def initialize(opts={}, synapse)
       # use the previous backends that we already know about.
       @use_previous_backends = opts.fetch('use_previous_backends', true)
 
+      # If backends discovered with this watcher lack ports or discovered
+      # ports should be ignored, this option indicates that
+      @backend_port_override = opts['backend_port_override'] || nil
+
+      # For backwards compatability we support server_port_override
+      # This will be removed in future versions
+      if @backend_port_override.nil? && @generator_config['haproxy']
+        @backend_port_override = @generator_config['haproxy']['server_port_override']
+      end
+
       # set a flag used to tell the watchers to exit
       # this is not used in every watcher
       @should_exit = false
 
       validate_discovery_opts
     end
 
+    def haproxy
+      log.warn "synapse: service watcher #{@name} accessing watcher.haproxy. This is DEPRECATED and will be removed in future iterations, use watcher.generator_config['haproxy'] instead."
```

Comment:
This was really just to help me when porting the service watchers. I think this would only impact folks if they had custom service watchers (all of the ones in the repo have been ported).